in-progress/8754-slashing-and staking.md

```diff
@@ -0,0 +1,357 @@
+
+In the happy path, the Executing Validators are able to obtain the `TxObjects` and proofs required to do so. If the resulting state root does not match what was published to L1, they read the list of all committee members who signed the invalid block. 
+
+The Executing Validator prepares a payload containing:
+1) The offense in question: Invalid state root
+2) The slot number of the invalid block.
+3) The list of committee members who signed off on the invalid block.
+
+and gossips through to the validator set. 
+
+Validators who receive this proposal first also read the L1 for the list of `TxHash` posted along the block in question. If it does not match what's in the slashing proposal, it discards the slashing proposal. 
+
+If the list of `TxHash` does match, they request all `TxEffects` and proofs from the p2p. As a last resort, they request from the validator who initiated the slashing proposal. They also read the state root posted to L1. 
+
+If they're able to retreive all the required data to re-xecute the block contents and find an invalid state root, they sign the slashing proposal and gossip the signed proposal via the p2p. 
+
+The executing validator who initialized the slashing proposal must aggregate signatures and post to L1 for verification once >50% of the stake has signed the proposal. 
+
+On the L1, the contract verifies that: 
+1) The epoch has indeed re-orged.
+2) The committee members named in the slashing proposal have signed the specific block in question.
+3) Validators in control of more than 50% of stake have signed the slashing proposal.
+
+If the above checks pass, the named committee members are slashed. The L1 contracts pays out a small fee to cover the cost of verifying signatures on the L1. 
+
+```mermaid
+sequenceDiagram
+title Slashing an invalid state root - Happy Path
+
+    participant A as Proposer
+    participant B as Committee
+    participant C as Validator Set
+    participant D as Executing Validator
+    participant E as Ethereum L1
+
+    par
+        A ->> B: Gossip proposal
+        A ->> C: Gossip proposal
+    end
+
+    par
+        B ->> A: Gossip Attestations
+        B ->> C: Gossip Attestations
+    end
+
+    A ->> E: Propose Block
+    D ->> E: Retrieve TxHash
+    par
+        D ->> B: Request TxEffect + Proofs
+        D ->> C: Request TxEffect + Proofs
+    end
+
+    D ->> D: Execute block and check state root on L1
+    
+    alt If state root matches L1
+        D ->> D: Do nothing
+    else If state root does not match L1
+        D ->> C: Send proposal to slash malicious signatures
+    end
+
+    C ->> E: Retrieve Tx Hash
+    C ->> D: Request TxEffect + Proofs
+    C ->> C: Execute Proposal
+    alt If state root matches L1
+        C ->> C: Do nothing
+    else If state root doesn't match L1
+        C ->> D: Sign slashing proposal
+    end
+    D ->> D: Aggregate BLS signatures
```

Comment:
bls has entered the chat

Reply:
At the moment without sufficient precompiles, there is a non negligible cost to producing the aggregate signatures. (additions over the BLS12-381 curve). Are we asssuming the existence of these precompiles to do this or will we do it anyway